server/services/courtAppearanceService.ts

```diff
@@ -17,6 +18,41 @@ export default class CourtAppearanceService {
   constructor() {}
 
   setCaseReferenceNumber(
+    session: CookieSessionInterfaces.CookieSessionObject,
+    nomsId: string,
+    courtCaseReferenceForm: CourtCaseReferenceForm,
+  ) {
+    const errors = validate(
+      courtCaseReferenceForm,
+      {
+        referenceNumber: [
+          'required_without:noCaseReference',
+          `onlyOne:${courtCaseReferenceForm.noCaseReference ?? ''}`,
+          'regex:/^[T]{1}\\d{8}$/',
```

Comment:
This may be too restrictive in the future but can be easily removed, it means the case reference must be T followed by 8 numbers